pkg/scheduler/objects/application.go

```diff
@@ -1638,6 +1640,7 @@ func (sa *Application) addAllocationInternal(info *Allocation) {
 					zap.Error(err))
 			}
 		}
+		sa.hasPlaceholderAlloc.Store(true)
```

Comment:
-1 that cannot work.

Applications at the partition level are a map of all applications in the partition. The queue that we are checking needs to make the decision. At the partition level we do not know if the specific queue the app is running in has pending requests. We also do not want to walk over all applications at the partition level as that would not take into account any queue details.

Filtering out that early does not work.

Reply:
If there's no application with an active placeholder allocation, then we don't have to call tryPlaceholderAllocate()` because there's no placeholder to replace. Obviously if this counter is >1 then we have to descend the hierarchy and see where can we can do a replacement.